### PR TITLE
[Feature][Core] Rename `result_table_name`/`source_table_name` to `plugin_input/plugin_output`

### LINKED ISSUE #8062: [Feature][Config] Rename `source_table_name `/`result_table_name `
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

I think `result_table_name` is ambiguous with `source_table_name` and needs to be redefined to a suitable name. Do you have any suggestions? 

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hailin0**
### Purpose of this pull request

close #8062

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by Hisoka-X** [comment]
This is a big change. cc @TyrantLucifer @liugddx @liunaijie @dailai @Carl-Zhou-CN

**Review by dailai — DISMISSED** [review]
LGTM

**Comment by liugddx** [comment]
Will this affect existing tasks?

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/common/CommonOptions.java [hunk 1]
```diff
@@ -30,28 +30,30 @@ public interface CommonOptions {
                     .noDefaultValue()
                     .withDescription("Name of the SPI plugin class.");
 
-    Option<String> RESULT_TABLE_NAME =
-            Options.key("result_table_name")
+    Option<String> PLUGIN_OUTPUT =
+            Options.key("plugin_output")
                     .stringType()
                     .noDefaultValue()
+                    .withFallbackKeys("result_table_name")
```
Comment (hailin0) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/common/CommonOptions.java:37]: Compatible with old versions

cc @liugddx

**Comment by Carl-Zhou-CN** [comment]
@hailin0  Do we need to keep a few test cases and end-to-end tests for backward compatibility with the old version？

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-fake-e2e/src/test/java/org/apache/seatunnel/e2e/connector/fake/FakeIT.java [hunk 2]
```diff
@@ -41,5 +41,9 @@ public void testFakeConnector(TestContainer container)
         Container.ExecResult fakeComplex =
                 container.executeJob("/fake_generic_row_type_to_assert.conf");
         Assertions.assertEquals(0, fakeWithTemplate.getExitCode());
+        Container.ExecResult compatibleTableNameCase =
+                container.executeJob(
+                        "/fake_to_assert_with_compatible_source_and_result_table_name.conf");
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-fake-e2e/src/test/java/org/apache/seatunnel/e2e/connector/fake/FakeIT.java:46]: add testcase for old versions

cc @Carl-Zhou-CN

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-fake-e2e/src/test/resources/fake_to_assert_with_compatible_source_and_result_table_name.conf [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+    schema = {
+      fields {
+        f1 = int
+        f2 = string
+      }
+    }
+  }
+  FakeSource {
+    result_table_name = "fake2"
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-fake-e2e/src/test/resources/fake_to_assert_with_compatible_source_and_result_table_name.conf:36]: add testcase for old versions 

cc @Carl-Zhou-CN

**Comment by hailin0** [comment]
PTAL

**Review by dailai — DISMISSED** [review]

**Review by liunaijie — DISMISSED** [review]

**Comment by liunaijie** [comment]
This is a big change, even it's backward compatibility with the old version, but how many version should we keep backward, all the time or like 2 or 3 versions?

How about add a upgrade note on each release version, in this page, introduce how to upgrade from old version (which is the minimum version).  which feature (parameters) is deprecated, moreover we can introduce the features, fixed bug. 

Now we have the release-node, but it's not enough.

WDYT

**Comment by Hisoka-X** [comment]
>  how many version should we keep backward, all the time or like 2 or 3 versions

I think we should keep it indefinitely, not just for 2 or 3 versions. Perhaps after one or two years, we can revisit it to decide whether to keep or remove it, as keeping it poses no burden to us. However, we should mention in the documentation that the use of result_table_name and source_table_name is no longer recommended. cc @hailin0 

> How about add a upgrade note on each release version, in this page, introduce how to upgrade from old version (which is the minimum version). which feature (parameters) is deprecated, moreover we can introduce the features, fixed bug.

+1

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-fake-e2e/src/test/java/org/apache/seatunnel/e2e/connector/fake/FakeIT.java (hunk 2 above)
Comment (Carl-Zhou-CN) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-fake-e2e/src/test/java/org/apache/seatunnel/e2e/connector/fake/FakeIT.java:46]: ok

**Comment by Carl-Zhou-CN** [comment]
> > how many version should we keep backward, all the time or like 2 or 3 versions
> 
> I think we should keep it indefinitely, not just for 2 or 3 versions. Perhaps after one or two years, we can revisit it to decide whether to keep or remove it, as keeping it poses no burden to us. However, we should mention in the documentation that the use of result_table_name and source_table_name is no longer recommended. cc @hailin0
> 
> > How about add a upgrade note on each release version, in this page, introduce how to upgrade from old version (which is the minimum version). which feature (parameters) is deprecated, moreover we can introduce the features, fixed bug.
> 
> +1

+1

**Comment by hailin0** [comment]
> This is a big change, even it's backward compatibility with the old version, but how many version should we keep backward, all the time or like 2 or 3 versions?
> 
> How about add a upgrade note on each release version, in this page, introduce how to upgrade from old version (which is the minimum version). which feature (parameters) is deprecated, moreover we can introduce the features, fixed bug.
> 
> Now we have the release-node, but it's not enough.
> 
> WDYT

Yes, this is what is still missing

**Comment by hailin0** [comment]
> > how many version should we keep backward, all the time or like 2 or 3 versions
> 
> I think we should keep it indefinitely, not just for 2 or 3 versions. Perhaps after one or two years, we can revisit it to decide whether to keep or remove it, as keeping it poses no burden to us. However, we should mention in the documentation that the use of result_table_name and source_table_name is no longer recommended. cc @hailin0
> 
> > How about add a upgrade note on each release version, in this page, introduce how to upgrade from old version (which is the minimum version). which feature (parameters) is deprecated, moreover we can introduce the features, fixed bug.
> 
> +1

@Hisoka-X 
Current, an outdated parameter warning will be printed in the log.

**Comment by Hisoka-X** [comment]
> Current, an outdated parameter warning will be printed in the log.

We should metion it in doc too, let users to consciously change expired configurations. Doc more important than log.

**Comment by hailin0** [comment]
> > Current, an outdated parameter warning will be printed in the log.
> 
> We should metion it in doc too, let users to consciously change expired configurations. Doc more important than log.

Added notes.

**Review by Hisoka-X — APPROVED** [review]

**Review by dailai — APPROVED** [review]

**Pull request merged**